pkg/detectors/privatekey/ssh.go

```diff
@@ -91,9 +103,23 @@ func sshDialWithContext(ctx context.Context, network, addr string, config *ssh.C
 		return nil, fmt.Errorf("error dialing %s: %w", addr, err)
 	}
 
+	var once sync.Once
+	closeConn := func() {
+		once.Do(func() {
+			conn.Close()
+		})
+	}
+
+	// Monitor the context and close the connection if canceled.
+	// This ensures that the connection is closed when the context is canceled, preventing leaks.
+	go func() {
+		<-ctx.Done()
+		closeConn()
+	}()
```

Comment:
Would moving the goroutine to after the error check for `ssh.NewClientConn` solve the double-close issue? Like around line 125.

Reply:
Ooo, why you sir are a genius. Thanks, that should do it. ❤️

Reply:
Oh no! I lied... looks like `ssh.NewClientConn` hangs. 😓